safeeyes/tests/test_core.py

```diff
@@ -0,0 +1,584 @@
+from collections import deque
+import datetime
+import logging
+import pytest
+
+from safeeyes import core
+from safeeyes import model
+
+import threading
+from time import sleep
+
+from unittest import mock
+
+class TestSafeEyesCore:
+    @pytest.fixture(autouse=True)
+    def set_time(self, time_machine):
+        time_machine.move_to(datetime.datetime.fromisoformat("2024-08-25T13:00:00+00:00"), tick=False)
+
+    @pytest.fixture(autouse=True)
+    def monkeypatch_translations(self, monkeypatch):
+        monkeypatch.setattr(core, "_", lambda message: "translated!: " + message, raising=False)
+        monkeypatch.setattr(model, "_", lambda message: "translated!: " + message, raising=False)
+
+    @pytest.fixture
+    def sequential_threading(self, monkeypatch, time_machine):
```

Comment:
To tell the truth, it is quite difficult to understand. Maybe move this to separate PR? It won't make reading easy but the PR scope will be more specific